src/systems/performer_detector/PerformerDetector.cc

```diff
@@ -0,0 +1,208 @@
+ * limitations under the License.
+ *
+ */
+
+#include <ignition/msgs/pose.pb.h>
+
+#include <ignition/common/Profiler.hh>
+#include <ignition/math/AxisAlignedBox.hh>
+#include <ignition/math/Vector3.hh>
+#include <ignition/plugin/Register.hh>
+#include <ignition/transport/Node.hh>
+#include <sdf/Box.hh>
+#include <sdf/Element.hh>
+#include <sdf/Geometry.hh>
+
+#include "ignition/gazebo/Model.hh"
+#include "ignition/gazebo/Util.hh"
+#include "ignition/gazebo/components/Geometry.hh"
+#include "ignition/gazebo/components/Name.hh"
+#include "ignition/gazebo/components/ParentEntity.hh"
+#include "ignition/gazebo/components/Performer.hh"
+#include "ignition/gazebo/components/Pose.hh"
+
+#include "PerformerDetector.hh"
+
+using namespace ignition;
+using namespace gazebo;
+using namespace systems;
+
+/////////////////////////////////////////////////
+void PerformerDetector::Configure(const Entity &_entity,
+               const std::shared_ptr<const sdf::Element> &_sdf,
+               EntityComponentManager &_ecm,
+               EventManager &/*_eventMgr*/)
+{
+  this->model = Model(_entity);
+  if (!this->model.Valid(_ecm))
+  {
+    ignerr << "PerformerDetector should be attached to a model entity. "
+           << "Failed to initialize." << std::endl;
+    return;
+  }
+
+  this->detectorName = this->model.Name(_ecm);
+
+  auto modelPose = _ecm.Component<components::Pose>(_entity)->Data();
+
+  auto sdfClone = _sdf->Clone();
+  bool hasGeometry{false};
+  if (sdfClone->HasElement("geometry"))
+  {
+    auto geom = sdfClone->GetElement("geometry");
+    if (geom->HasElement("box"))
+    {
+      auto box = geom->GetElement("box");
+      auto boxSize = box->Get<math::Vector3d>("size");
+      this->detectorGeometry = math::AxisAlignedBox(-boxSize / 2, boxSize / 2);
+      hasGeometry = true;
+    }
+  }
+
+  if (!hasGeometry)
+  {
+    ignerr << "'<geometry><box>' is a required parameter for "
+              "PerformerDetector. Failed to initialize.\n";
+    return;
+  }
+
+  std::string defaultTopic{"/model/" + this->model.Name(_ecm) +
+                             "/performer_detector/status"};
+  auto topic = _sdf->Get<std::string>("topic", defaultTopic).first;
+
+  ignmsg << "PerformerDetector publishing messages on "
+         << "[" << topic << "]" << std::endl;
+
+  transport::Node node;
+  this->pub = node.Advertise<msgs::Pose>(topic);
+  this->initialized = true;
+}
+
+//////////////////////////////////////////////////
+void PerformerDetector::PostUpdate(
+  const ignition::gazebo::UpdateInfo &_info,
+  const ignition::gazebo::EntityComponentManager &_ecm)
+{
+  IGN_PROFILE("PerformerDetector::PostUpdate");
+
+  if (_info.paused)
+    return;
+
+  if (this->initialized)
+  {
+    auto modelPose =
+        _ecm.Component<components::Pose>(this->model.Entity())->Data();
+
+    // Double negative because AxisAlignedBox does not currently have operator+
+    // that takes a position
```

Comment:
:scream: https://github.com/ignitionrobotics/ign-math/issues/121

Reply:
Thanks for creating the issue